src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser class deals with making sense of the user command.
+ */
+public class Parser {
+    /**
+     * Parses the command typed by the user.
+     *
+     * @param str the string input by the user
+     * @param tasks the TaskList
+     * @throws DukeException if command is invalid
+     */
+    public static void parseCommand(String str, TaskList tasks) throws DukeException {
+        String[] splitStr = str.split(" ", 2);
+        switch (splitStr[0]) {
```

Comment:
I like your use of `switch` statement here, it makes your code very neat and readable 😃